packages/onboard-react/src/components/WalletDialog/WalletDialog.tsx

```diff
@@ -1,48 +1,109 @@
-import React, { FC, useCallback, useMemo, useState } from "react";
+import React, { FC, useCallback, useEffect, useMemo, useState } from "react";
+
 import {
   Dialog,
   VerticalTabs,
   VerticalTabsItem,
+  VerticalTabsMobile,
   Button,
   ButtonStyle,
   Heading,
   DialogSize,
 } from "@sovryn/ui";
 import { connectWallet$ } from "@sovryn/onboard-core/dist/streams";
+import resolveConfig from "tailwindcss/resolveConfig";
+import tailwindConfig from "@sovryn/tailwindcss-config";
+
 import { FilterType, WalletList } from "../WalletList/WalletList";
 import { InstructionsTab } from "../InstructionsTab/InstructionsTab";
 import { HardwareWallets } from "../HardwareWallets/HardwareWallets";
 import styles from "./WalletDialog.module.css";
+import { ButtonBack } from "../ButtonBack/ButtonBack";
+import { useSubscription } from "../../hooks/useSubscription";
+import { selectAccounts$ } from "@sovryn/onboard-hw-common";
+
+const config = resolveConfig(tailwindConfig);
 
 type WalletDialogProps = {
   isOpen: boolean;
 };
 
 const WalletDialog: FC<WalletDialogProps> = ({ isOpen }) => {
+  const { inProgress } = useSubscription(selectAccounts$);
+  const [isMobile, setIsMobile] = useState(false);
   const [index, setIndex] = useState(0);
+  const [indexMobile, setIndexMobile] = useState<null | number>(null);
+
+  const onChangeIndex = useCallback((index: number | null) => {
+    index !== null ? setIndex(index) : setIndex(0);
+    setIndexMobile(index);
+  }, []);
+
+  const getPlatformType = useCallback(() => {
```

Comment:
I would extract all of the mobile detection functionality into a custom hook (e.g. `useIsMobile`), that would just return `isMobile`. The reason is, it will simplify the component + we can reuse it elsewhere in the future if we need to.